pkg/querier/vcs/service.go

```diff
@@ -31,44 +33,96 @@ func New(logger log.Logger) *Service {
 
 func (q *Service) GithubApp(ctx context.Context, req *connect.Request[vcsv1.GithubAppRequest]) (*connect.Response[vcsv1.GithubAppResponse], error) {
 	return connect.NewResponse(&vcsv1.GithubAppResponse{
-		ClientID: client.GithubAppClientID,
+		ClientID: githubAppClientID,
 	}), nil
 }
 
 func (q *Service) GithubLogin(ctx context.Context, req *connect.Request[vcsv1.GithubLoginRequest]) (*connect.Response[vcsv1.GithubLoginResponse], error) {
-	resp := connect.NewResponse(&vcsv1.GithubLoginResponse{})
-	cookie, err := client.AuthorizeGithub(ctx, req.Msg.AuthorizationCode)
+	cfg, err := githubOAuthConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to authorize github: %w", err)
+		q.logger.Log("err", err, "msg", "failed to get GitHub OAuth config")
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to authorize with GitHub"))
```

Comment:
This pattern of "log the real error, return a generic message" is so we don't accidentally leak anything sensitive to the UI. While errors won't include tokens or any really sensitive info, better to be on the safe side with what we send back to the client.